runtime/lua/vim/filetype.lua

```diff
@@ -29,8 +29,73 @@ local function getline(bufnr, start_lnum, end_lnum)
   return table.concat(lines) or ''
 end
 
--- Filetypes based on file extension
+---@private
+--- Get a single line or line-range from the buffer.
+---
+---@param bufnr number|nil The buffer to get the lines from
+---@param start_lnum number The line number of the first line (inclusive, 1-based)
+---@param end_lnum number|nil The line number of the last line (inclusive, 1-based)
+---@return table<string>|string Array of lines, or string when end_lnum is omitted
+function M.getlines(bufnr, start_lnum, end_lnum)
```

Comment:
We can remove the local `getline` function in favor of this one.

Reply:
Yes, I think I'll do that in the refactoring PR :+1: